### ISSUE TITLE
pacman: package groups' state always considered absent

### ISSUE THREAD

**Issue by hannahdiels** [ansible/ansible-modules-extras#766]
**Issue Type:** Bug Report

**Ansible Version:**

```
ansible 1.9.2
  configured module search path = None
```

**Ansible Configuration:** Not relevant

**Environment:** Arch Linux: Linux 4.1.2-2-ARCH #1 SMP PREEMPT x86_64 GNU/Linux

**Summary:**

No distinction is made between package groups and packages. Package group state=present is not detected, e.g. pacman -Qi base-devel will always report base-devel missing, as it checks for a package.

Relevant commands:

```
pacman -Sg base-devel  # print all packages in group
pacman -Qg base-devel  # print all packages in group that are installed
(pacman -Qg base-devel & pacman -Sg base-devel) | sort | uniq -u  # print packages in group that are not installed
```

My **suggested fix** is to start the module by building a list of packages to install/remove/whatever:
1. Determine whether {{ name }} is a group or package: `pacman -Sg base-devel` has `return code == 0` if it's a group, `1` otherwise.
2. A. If it's a group: pkgs = second column of the output of step 1.
   B. Else, pkgs is singleton of {{ name }}

You'll need to expand on the above to allow `name=a,b` and such.

**Steps To Reproduce:**

Put this task in a playbook, then run it twice:

```
- name: Install base-devel
  pacman: name=base-devel state=present
```

**Expected Results:**

Assuming base-devel wasn't installed, the first run would install all packages of base-devel. The second run would do nothing. (Ideally, only packages in base-devel that are absent will be installed; no reinstalls)

**Actual Results:**

All packages of base-devel are (re)installed, regardless of whether they were already present.

**Comment by rightaway** [comment]
+1

**Comment by dcbishop** [comment]
+1 from me to. It's kind of painful trying to install xorg for example. I basically have to specify 72 individual packages and that's not maintainable since it might change in the future.

**Comment by suizman** [comment]
Same here install xfce4 always show changes:

```
PLAY [localhost] ************************************************************** 
GATHERING FACTS *************************************************************** 
ok: [localhost]

TASK: [roles/kickass-xfce | Install base package Arch Linux] ****************** 
changed: [localhost]

TASK: [roles/kickass-xfce | Install extra package Arch Linux] ***************** 
ok: [localhost] => (item=xfwm4-themes)
ok: [localhost] => (item=numix-themes)

TASK: [roles/kickass-xfce | Install base package Debian/Ubuntu] *************** 
skipping: [localhost]

TASK: [roles/kickass-xfce | Install extra package Debian/Ubuntu] ************** 
skipping: [localhost]

TASK: [roles/kickass-xfce | Install base package CentOS/Fedora/SuSe] ********** 
skipping: [localhost]

TASK: [roles/kickass-xfce | Install extra package CentOS/Fedora/SuSe] ********* 
skipping: [localhost]

TASK: [roles/kickass-xfce | Copy xfce terminalrc file] ************************ 
ok: [localhost]

PLAY RECAP ******************************************************************** 
localhost                  : ok=4    changed=1    unreachable=0    failed=0   
```

In my case i am using Ansible 1.9.4 on Arch Linux..

**Comment by r-cheologist** [comment]
+1
This is killing me (my ssd).

**Comment by wkeeling** [comment]
This issue should now be resolved by #1651
